### DISCUSSION TITLE
Does Prisma Support Deeply Nested Sorting in One-to-Many Relationships?

### DISCUSSION THREAD

**Discussion by kristandy** [prisma/prisma#24702] [Q&A]
### Question

Hi, 

I have three schemas that are related to each other, and I want to sort by name in mme_role through mme_user. However, there seems to be an error when I try to do so. Does Prisma support deeply nested sorting in one-to-many relationship tables?

error
[1] prisma/user/user.ts(159,17): error TS2322: Type '{ user_roles: { role: { name: string; }; }; }' is not assignable to type 'Enumerable<mme_userOrderByWithRelationInput> | undefined'.
[1]   Types of property 'user_roles' are incompatible.
[1]     Object literal may only specify known properties, and 'role' does not exist in type 'mme_user_roleOrderByRelationAggregateInput'.

```prisma
model mme_role {
  id         Int             @id @default(autoincrement())
  name       String          @unique @db.VarChar(255)
  privileges mme_privilege[]
  user_roles mme_user_role[]
}

model mme_user_role {
  id      Int      @id @default(autoincrement())
  user_id Int
  user    mme_user @relation(fields: [user_id], references: [id], onDelete: Cascade, onUpdate: Cascade)
  role_id Int
  role    mme_role @relation(fields: [role_id], references: [id], onDelete: Cascade, onUpdate: Cascade)
}

model mme_user {
  id                 Int                      @id @default(autoincrement())
  userid             String                   @unique(map: "idxUser") @db.VarChar(25)
  passwd             String                   @db.VarChar(250)
  nama               String                   @db.VarChar(50)
  role               String                   @db.VarChar(15)
  email              String                   @db.VarChar(250)
  created_at         DateTime                 @db.DateTime(0)
  last_updated       DateTime                 @db.DateTime(0)
  is_deleted         Int                      @default(0) @db.UnsignedTinyInt
  deleted_at         DateTime?                @db.DateTime(0)
  logged_in          Int                      @default(0) @db.UnsignedTinyInt
  activities         mme_activity[]
  notifications      mme_notification[]
  product_price_list mme_product_price_list[]
  user_roles         mme_user_role[]

  @@index([is_deleted], map: "mme_user_is_deleted")
}

```

```ts
return prisma.mme_user.findMany({
        include: {
            user_roles: {
                include: {
                    role: true
                }
            }
        },        
        orderBy: {
            user_roles: {
                role : {
                    name: 'asc'
                }
            }
        }
    })
```

- OS: [Windows]
- Database: [MySQL]
- Node.js version: [22.3.0]
- Prisma: 4.16.2

**Comment by ludralph** [comment]
Hi @kristandy 👋 

Thank you for raising this question.

Prisma Client currently does not support sorting on deeply nested relations directly. We do have an open [issue](https://github.com/prisma/prisma/issues/13662) that aims to solve for this. It would be helpful if you could document your use case in this issue and also upvote it by leaving a 👍 to help our engineers prioritize it. As a workaround, you could use [raw query](https://www.prisma.io/docs/orm/prisma-client/queries/raw-database-access/raw-queries#queryraw) to implement this and leverage [SafeQL](https://www.prisma.io/docs/orm/prisma-client/queries/raw-database-access/custom-and-type-safe-queries#what-is-safeql) to make the raw query typesafe.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
